File: S2-Regression/Pt5-DecisionTreeRegression/decision_tree_regression.py
## Importing the libraries
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd


## Importing the data set
dataset = pd.read_csv('Position_Salaries.csv')
x = dataset.iloc[:, 1:-1].values
y = dataset.iloc[:, -1].values


## Training the Decision Tree Regression model on the whole data set
from sklearn.tree import DecisionTreeRegressor
regressor = DecisionTreeRegressor(random_state=0)
regressor.fit(x, y)


## Predicting a new result
y_pred = regressor.predict([[6.5]])


## Visualizing the Decision Tree Regression results (higher resolution)
x_grid = np.arange(min(x), max(x), .1)
x_grid = x_grid.reshape(len(x_grid), 1)
plt.scatter(x, y, color='red')
plt.plot(x_grid, regressor.predict(x_grid), color='blue')
plt.title('Truth or Bluff (Decision Tree Regression)')
plt.xlabel('Position Level')
plt.ylabel('Salary')
plt.show()

File: S2-Regression/Pt5-DecisionTreeRegression/decision_tree_regression.R
## Importing the data set
dataset = read.csv('Position_Salaries.csv')
dataset = dataset[2:3]

## Fitting Decision Tree Regression into the data set
# install.packages('rpart')
library('rpart')
regressor = rpart(formula = Salary ~ ., 
                  data = dataset, control = 
                    rpart.control(minsplit = 1))

## Predicting a new result
y_pred = predict(regressor, data.frame(Level = 6.5))

## Visualizing the Decision Tree Regression results
x_grid = seq(min(dataset$Level), max(dataset$Level), 0.01)
# install.packages('ggplot2')
library('ggplot2')
ggplot() +
  geom_point(aes(dataset$Level, dataset$Salary), colour = 'red') +
  geom_line(
    aes(x_grid, predict(regressor, newdata = data.frame(Level = x_grid))), 
    colour = 'blue'
  ) +
  ggtitle('Truth of Bluff (Decision Tree Regression)') +
  xlab('Position Level') +
  ylab('Salary')
